src/core/core/model/news_item_tag.py

```diff
@@ -123,22 +119,17 @@ def _parse_dict_tags(cls, tags: dict) -> dict[str, "NewsItemTag"]:
                 name = tag_key
                 tag_type = tag_data
             else:
-                name = tag_key
-                tag_type = "misc"
-
+                raise ValueError(f"Invalid tag format for key '{tag_key}': {type(tag_data).__name__} - must be str or dict")
+                
             parsed_tags[name] = NewsItemTag(name=name, tag_type=tag_type)
 
         return parsed_tags
 
     @classmethod
     def _parse_list_tags(cls, tags: list) -> dict[str, "NewsItemTag"]:
-        new_tags = {}
-        for tag in tags:
-            if isinstance(tag, dict):
-                tag_name = tag.get("name")
-                tag_type = tag.get("tag_type", "misc")
-            else:
-                tag_name = tag
-                tag_type = "misc"
-            new_tags[tag_name] = NewsItemTag(name=tag_name, tag_type=tag_type)
-        return new_tags
+        dict_tags = {
+            tag["name"]: tag
+            for tag in tags
+            if isinstance(tag, dict) and "name" in tag
+        }
```

Comment:
So we could either continue to accept those: [would be my suggestion]

```suggestion
        dict_tags = {}
        for tag in tags:
            if isinstance(tag, dict) and "name" in tag:
                dict_tags["name"] = tag
            elif isinstance(tag, str):
                dict_tags[tag] = {"name": tag, "tag_type": "misc"}
```

Or we just drop the string only from the tests and discontinue those.

Reply:
> So we could either continue to accept those: [would be my suggestion]

+1 to this one - but if feasible, could be also discontinued